proto/cosmos/base/abci/v1beta1/abci.proto

```diff
@@ -115,7 +123,13 @@ message MsgData {
 message TxMsgData {
   option (gogoproto.stringer) = true;
 
-  repeated MsgData data = 1;
+  // data field is deprecated and not populated.
+  repeated MsgData data = 1 [deprecated = true];
+
+  // msg_responses contains the Msg handler responses type packed in Anys.
+  //
+  // Since: cosmos-sdk 0.45
+  repeated google.protobuf.Any msg_responses = 2;
```

Comment:
cc @fdymylja, here's a proposal for solving #10496

Reply:
LGTM.

Reply:
How about:

```suggestion
  // data field is deprecated and not populated.
  repeated MsgData data = 1 [deprecated = true];

  // msg_responses contains the Msg handler responses type packed in Anys.
  //
  // Since: cosmos-sdk 0.45
  repeated bytes msg_responses = 2;
```

The bytes would be proto.Marshal(MsgXResponse), i.e. the Any.Value.

Pros: less space on disk; no need to create a `MsgResponse` interface to represent those Anys
Cons: not straight-forward, clients need to know/learn about this